src/contracts/libraries/SafetyBits.sol

```diff
@@ -83,16 +81,13 @@ library SafetyBits {
             | 1 << (SAFETY_LEVEL_OFFSET + uint16(SolverSafety.Unset))
     );
 
-    //TODO change to pure after testing
-    function getCurrentExecutionPhase(uint16 lockState) internal view returns (uint16) {
-        console.log("IN getCurrentExecutionPhase");
+    function getCurrentExecutionPhase(uint16 lockState) internal pure returns (uint16) {
+        // Isolate the middle 8 bits - only the ExecutionPhase bits
         uint16 isolatedPhaseBits = lockState & ONLY_EXECUTION_PHASE_MASK;   
-        console.log("isolatedPhaseBits: %s", TestUtils.uint16ToBinaryString(isolatedPhaseBits));
-        console.log("returning: %s", TestUtils.uint16ToBinaryString(uint16(isolatedPhaseBits >> EXECUTION_PHASE_OFFSET)));
-
-        // TODO Need to take dec number returned below and calculate log_2(dec num) to get the bit number
-
-        return uint16(isolatedPhaseBits >> EXECUTION_PHASE_OFFSET);
+        // Shift those 8 bits to the right, then convert the decimal value of those bits to UD60x18 form
+        UD60x18 phaseBitsInDecToBeLogged = ud((isolatedPhaseBits >> EXECUTION_PHASE_OFFSET) * SCALE);
+        // Calculate log2 of the dec value of phase bits above, scale back down, and convert to uint16
+        return uint16(phaseBitsInDecToBeLogged.log2().intoUint256() / SCALE);
```

Comment:
The log2() method here runs conversions, uses a for loop, and has all sorts of math.

In this case, since there aren't too many different possible execution phases, I'm sure that just running a 
if ("lock state & (1 << EXECUTION_PHASE_OFFSET + phaseWeWantToCheck) != 0) { return phaseWeWantToCheck}

and just run the bool check for the different phases until we find the right one.  
We could potentially even use a switch in assembly to save gas here too once we're comfy locking in the phases and the offsets (which I think we are at this point).

Reply:
Yeah, was just thinking something like a switch of if/else chain would probably be the most gas-efficient way to do it, as we only have 8 options in the enum and only some of them would be viable for donations. Will update this function to that approach